xdsl/utils/hints.py

```diff
@@ -2,45 +2,58 @@
 from types import UnionType
 from typing import (Annotated, Any, TypeGuard, TypeVar, Union, cast, get_args,
                     get_origin)
+from xdsl.ir import ParametrizedAttribute
+
+from xdsl.utils.exceptions import VerifyException
 
 _T = TypeVar("_T")
 
 
-def is_satisfying_hint(arg: Any, hint: type[_T]) -> TypeGuard[_T]:
+def isa(arg: Any, hint: type[_T]) -> TypeGuard[_T]:
```

Comment:
I feel `isa` is not a good name. Would you propose any alternatives?

Reply:
`isa` was the name that is currently use in MLIR, so that would make it easier for people coming from MLIR to understand it. I personally like it, because it reads like `isa([0], list[int])` like `[0] is a list[int]`.

Otherwise this function is a superset `isinstance`, so something like `is_of_type`?

Reply:
I'm wondering what the others think about it

Reply:
I think the MLIR argument makes a lot of sense, though I am wondering where we should draw the line. 

I also like it, as it reads quite intutively, IMO.